src/main/java/io/github/robwin/circuitbreaker/internal/RingBitSet.java

```diff
@@ -48,6 +48,17 @@ public RingBitSet(int bitSetSize) {
         bitSet = new BitSetMod(bitSetSize);
     }
 
+    public RingBitSet(int bitSetSize, RingBitSet sourceSet) {
+        this(bitSetSize);
+
+        int targetLength = Integer.min(bitSetSize, sourceSet.length);
+        int sourceIndex = sourceSet.index;
+        for (int i = 0; i < targetLength; i++) {
```

Comment:
I think this should be in reverse order  so that we copy the latest call results. Or?